packages/steiger/src/models/config/validate-config.ts

```diff
@@ -1,8 +1,16 @@
 import z from 'zod'
 
+import { BaseRuleOptions, Config, Plugin, Severity } from '@steiger/types'
+
 import { getOptions, isConfigObject, isPlugin } from './raw-config'
 import { isEqual } from '../../shared/objects'
-import { BaseRuleOptions, Config, Plugin, Severity } from '@steiger/types'
+
+const OLD_CONFIG_ERROR_MESSAGE =
+  'Old configuration structure noticed. We are evolving!\nPlease follow these easy 5-minute instructions to migrate to the new one:\nhttps://github.com/feature-sliced/steiger/blob/master/MIGRATION_GUIDE.md'
+const WRONG_CONFIG_SHAPE_ERROR_MESSAGE =
+  'The provided config has a wrong shape.\nPlease check your configuration file and follow the documentation to fix it:\nhttps://github.com/feature-sliced/steiger?tab=readme-ov-file#configuration'
```

Comment:
suggestion: reworded the messages slightly. Some changes are because it sounds more natural to me that way, others, like the migration guide, I felt we should change because we don't get to decide if the guide is easy, or that it takes 5 minutes :)

```suggestion
const OLD_CONFIG_ERROR_MESSAGE =
  'Old configuration format detected. We are evolving!\nPlease follow this little guide to migrate to the new one:\nhttps://github.com/feature-sliced/steiger/blob/master/MIGRATION_GUIDE.md'
const WRONG_CONFIG_SHAPE_ERROR_MESSAGE =
  'The config should be an Array, but the provided config is not.\nHere is a link to the documentation that might help to fix it:\nhttps://github.com/feature-sliced/steiger?tab=readme-ov-file#configuration'
```